index.bs

```diff
@@ -583,36 +583,30 @@ interface ML {
 };
 </script>
 
-The {{ML/createContext()}} method steps are:
-1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, then throw a "{{SecurityError!!exception}}" {{DOMException}} and abort these steps.
-1. Let |promise| be [=a new promise=].
-1. Let |context| be a new {{MLContext}} object.
-1. Switch on the method's first argument:
-    <dl class=switch>
-    <dt>{{MLContextOptions}}
-    <dd>Set |context|.{{[[contextType]]}} to [=default-context|default=].
-    <dd>Set |context|.{{[[deviceType]]}} to the value of {{MLContextOptions}}'s {{deviceType}}.
-    <dd>Set |context|.{{[[powerPreference]]}} to the value of {{MLContextOptions}}'s {{powerPreference}}.
-
-    <dt>{{GPUDevice}}
-    <dd>Set |context|.{{[[contextType]]}} to [=webgpu-context|webgpu=].
-    <dd>Set |context|.{{[[deviceType]]}} to "[=device-type-gpu|gpu=]".
-    <dd>Set |context|.{{[[powerPreference]]}} to "[=power-preference-default|default=]".
-    </dl>
-1. Issue the following steps to a separate timeline:
-    1. If the User Agent can support the |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}, then:
-        1. Set |context|.{{MLContext/[[implementation]]}} to an implementation supporting |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}.
-        1. [=Resolve=] |promise| with |context|.
-    1. Else:
-        1. [=Resolve=] |promise| with a new {{NotSupportedError}}.
-1. Return |promise|.
-
 ### Permissions Policy Integration ### {#permissions-policy-integration}
 
 This specification defines a <a>policy-controlled feature</a> identified by the
 string "<code><dfn data-lt="webnn-feature">webnn</dfn></code>".
 Its <a>default allowlist</a> is <code>'self'</code>.
 
+### The {{ML/createContext()}} method ### {#api-ml-createcontext}
+The {{ML/createContext()}} method steps are:
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, return [=a new promise=] [=rejected=] with a "{{SecurityError}}" and abort these steps.
+1. Let |promise| be [=a new promise=].
+1. Return |promise| and run the following steps [=in parallel=].
+1. Let |context| be a new {{MLContext}} object.
+1. Let |options| be the first argument.
+1. If |options| is a {{GPUDevice}},
```

Comment:
@anssiko could you check on this, too? are we allowed to say "is a <object-defined-here>", or just "is an object" and specify prose? I suspect the latter :)

Reply:
I think this would be the most explicit:

`1. If options is a GPUDevice object, then`

Reply:
In that case we don't need the `object`, and other specs are not using it, either. 
We can leave it as is for now, and make a sweep-check later on this topic.

Reply:
“is a Foo object” makes the type explicit without the need to click the link.

See also https://www.w3.org/TR/webgpu/#dom-gpuadapter-requestdevice

Reply:
I see both patterns in different specs (infra, url, html, etc)
 - if an object  or variable is defined in the spec or algo, just use it by name
 - generic object, list, promise etc are referred as "a list", "an object" etc.
or,
- use the type next to the name.
If you prefer having the type mentioned, we can follow that convention. I used to prefer brevity when clarity is not affected.

Reply:
I think including the type too would be better.

I usually check the mothership HTML LS for conventions because it is well maintained. It looks like "is a Foo object" is the convention in algorithms:

https://html.spec.whatwg.org/#media-element-load-algorithm
https://html.spec.whatwg.org/#convert-to-a-list-of-name-value-pairs
https://html.spec.whatwg.org/#font-style-source-object
https://html.spec.whatwg.org/#dom-path2d
https://html.spec.whatwg.org/#check-the-usability-of-the-image-argument
https://html.spec.whatwg.org/#dom-context-2d-fillstyle
https://html.spec.whatwg.org/#dom-context-2d-strokestyle

Also, I had this type of hypothetical test in mind:
```
async function getGPUDevice() {
  const adapter = await navigator.gpu.requestAdapter();
  if (!adapter) { return false; }
  const device = await adapter.requestDevice();
  return device;
}

function isGPUDevice(device) {
  return device instanceof GPUDevice;
}

function isObject(device) {
  return device instanceof Object;
}

const device = await getGPUDevice();
console.log(isGPUDevice(device));
console.log(isObject(device));
```

Both the assertions should return true, thus "is a GPUDevice object" prose more explicitly maps to these two test assertions.

Reply:
Yes, the current text is 

> If |options| is a {{GPUDevice}} object

Reply:
LGTM